traversal/procedure/ProcedureVertex.java

```diff
@@ -142,11 +155,20 @@ public Thing asThing() {
             return this;
         }
 
-        public Set<ProcedureEdge<?, ?>> inRolePlayers() {
-            if (inRolePlayers == null) {
-                inRolePlayers = iterate(ins()).filter(ProcedureEdge::isRolePlayer).toSet();
+        public boolean isScope() {
+            if (isScope == null) {
+                isScope = iterate(outs()).anyMatch(e -> e.direction().isForward() && (e.isRolePlayer() || e.isRelating())) ||
+                        iterate(ins()).anyMatch(e -> e.direction().isBackward() && (e.isRolePlayer() || e.isRelating()));
+            }
+            return isScope;
+        }
```

Comment:
a vertex with `RolePlayer` or `Relating` edges in certain directions mandates that the vertex is a Scope, ie. only generates Relations.